### DISCUSSION TITLE
Asyncify: add generators support

### DISCUSSION THREAD

**Discussion by MarkParker5** [fastapi/asyncer#86] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Asyncer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Asyncer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Asyncer but to [AnyIO](https://github.com/agronholm/anyio).
- [X] I already checked if it is not related to Asyncer but to [Trio](https://github.com/python-trio/trio).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import time
from asyncer import asyncify, runnify


def sync_generator(): # generator with different long sync blocking code between yields
    for i in range(10):
        time.sleep(5) # random long sync blocking code
        yield i

async def main():
    async for i in asyncify(sync_generator)(): # requires an object with __aiter__ method, got coroutine
        print('->', i)

runnify(main)()
```

### Description

I love asyncer and use it in several of my projects. It would be great to be able to use `asyncify` not only for functions, but also for generators, which are essentially the same functions, but with multiple returns.

### Operating System

Other

### Operating System Details

Any

### asyncer Version

0.0.2

### Python Version

3.10

**Comment by WojciechMigda** [comment]
I concur. It would come handy for streaming in Flask (passing syncified async generator to `flask.Response`).
